### PR TITLE
Create release pipeline as a yaml pipeline

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Convert the visual release pipeline to a yaml pipeline

Added a screenshot of the trial run.

image

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: tools/releaseBuild/azureDevOps/templates/release-CreateGitHubDraft.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+steps:
+- download: none
+
+- template: release-SetReleaseTagAndContainerName.yml
+
+- pwsh: |
+    $azcopy = "C:\Program Files (x86)\Microsoft SDKs\Azure\AzCopy\AzCopy.exe"
```
Comment (TravisEz13) [inline tools/releaseBuild/azureDevOps/templates/release-CreateGitHubDraft.yml:7]: I think we should move this to macos and use brew to make sure the tools are there.  This just assumes.

Comment (TravisEz13) [reply tools/releaseBuild/azureDevOps/templates/release-CreateGitHubDraft.yml:7]: not blocking

File: tools/releaseBuild/azureDevOps/templates/release-MakeContainerPublic.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+steps:
+- download: none
+
+- template: release-SetReleaseTagAndContainerName.yml
+
+- pwsh: |
+    az login --service-principal -u $(az_url) -p $(az_key) --tenant $(az_name)
```
Comment (TravisEz13) [inline tools/releaseBuild/azureDevOps/templates/release-MakeContainerPublic.yml:7]: We should declare what variable group we depend on for this job

File: tools/releaseBuild/azureDevOps/templates/release-PublishPackageMsftCom.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+- pwsh: |
+   git clone https://$(AzureDevOpsPat)@mscodehub.visualstudio.com/PowerShellCore/_git/Internal-PowerShellTeam-Tools '$(Pipeline.Workspace)/tools'
+  displayName: Clone Internal-PowerShellTeam-Tools from MSCodeHub
+
+- download: releasePipeline
+  artifact: metadata
+
+- pwsh: |
+    $metadata = Get-Content -Path "$ENV:PIPELINE_WORKSPACE/releasePipeline/metadata/release.json" -Raw | ConvertFrom-Json
+    Write-Verbose -Verbose "$(releaseTag) $(PackagesRepoPassword) $(AzureVersion) $($metadata.LTSRelease)"
```
Comment (TravisEz13) [inline tools/releaseBuild/azureDevOps/templates/release-PublishPackageMsftCom.yml:20]: declare what variable group we depend on

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
We should declare at the job level, what variable groups they depend on
https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#variables

File: tools/releaseBuild/azureDevOps/templates/release-CreateGitHubDraft.yml (hunk 1 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/templates/release-CreateGitHubDraft.yml:7]: It works currently, i think we should keep it as is. The agent guarantees AzCopy to be present: https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md#cli-tools

File: tools/releaseBuild/azureDevOps/templates/release-MakeContainerPublic.yml (hunk 2 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/templates/release-MakeContainerPublic.yml:7]: Added variables group in all jobs that use variables from groups

Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/templates/release-MakeContainerPublic.yml:7]: fixed

File: tools/releaseBuild/azureDevOps/templates/release-PublishPackageMsftCom.yml (hunk 3 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/templates/release-PublishPackageMsftCom.yml:20]: fixed

**Review by TravisEz13 — APPROVED** [review]

**Review by daxian-dbw — APPROVED** [review]

**Review by rjmholt — APPROVED** [review]

**Pull request merged**